Panic parsing logparser config

## Bug report

### Relevant telegraf.conf:

```
[[inputs.logparser]]
	[inputs.logparser.grok]
		patterns = [""]
```

### System info:

Current master (6e76731)

### Steps to reproduce:

1. run telegraf

### Expected behavior:
No error, or graceful error

### Actual behavior:
```
2017-04-30T17:40:18Z I! Starting Telegraf (version )
2017-04-30T17:40:18Z I! Loaded outputs: file
2017-04-30T17:40:18Z I! Loaded inputs: inputs.logparser
2017-04-30T17:40:18Z I! Tags enabled: host=whistler
2017-04-30T17:40:18Z I! Agent Config: Interval:10s, Quiet:false, Hostname:"whistler", Flush Interval:10s 
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/influxdata/telegraf/plugins/inputs/logparser/grok.(*Parser).addCustomPatterns(0xc4201e4580, 0xc42028b790)
	/tmp/.gdm_15715/src/github.com/influxdata/telegraf/plugins/inputs/logparser/grok/grok.go:293 +0x14d
github.com/influxdata/telegraf/plugins/inputs/logparser/grok.(*Parser).Compile(0xc4201e4580, 0xc4202d8500, 0x199)
	/tmp/.gdm_15715/src/github.com/influxdata/telegraf/plugins/inputs/logparser/grok/grok.go:133 +0x91a
github.com/influxdata/telegraf/plugins/inputs/logparser.(*LogParserPlugin).Start(0xc4202d8500, 0x1afd540, 0xc42000c600, 0x0, 0x0)
	/tmp/.gdm_15715/src/github.com/influxdata/telegraf/plugins/inputs/logparser/logparser.go:122 +0x4f6
github.com/influxdata/telegraf/agent.(*Agent).Run(0xc4201420d0, 0xc42013e900, 0x0, 0x0)
	/tmp/.gdm_15715/src/github.com/influxdata/telegraf/agent/agent.go:352 +0x658
main.reloadLoop(0xc42013e600, 0x1c096d0, 0x0, 0x0, 0x1c096d0, 0x0, 0x0, 0x1c096d0, 0x0, 0x0, ...)
	/home/phemmer/.go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:222 +0xa06
main.main()
	/home/phemmer/.go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:392 +0x65b
```

--- a/plugins/inputs/logparser/grok/grok.go
+++ b/plugins/inputs/logparser/grok/grok.go
@@ -118,11 +118,18 @@ func (p *Parser) Compile() error {
 
 	// Give Patterns fake names so that they can be treated as named
 	// "custom patterns"
-	p.namedPatterns = make([]string, len(p.Patterns))
+	p.namedPatterns = make([]string, 0, len(p.Patterns))
 	for i, pattern := range p.Patterns {
+		if pattern == "" {
+			continue
+		}
 		name := fmt.Sprintf("GROK_INTERNAL_PATTERN_%d", i)
 		p.CustomPatterns += "\n" + name + " " + pattern + "\n"
-		p.namedPatterns[i] = "%{" + name + "}"
+		p.namedPatterns = append(p.namedPatterns, "%{"+name+"}")
+	}
+
+	if len(p.namedPatterns) == 0 {
+		return fmt.Errorf("pattern required")
 	}
 
 	// Combine user-supplied CustomPatterns with DEFAULT_PATTERNS and parse

--- a/plugins/inputs/logparser/logparser.go
+++ b/plugins/inputs/logparser/logparser.go
@@ -117,15 +117,10 @@ func (l *LogParserPlugin) Start(acc telegraf.Accumulator) error {
 	}
 
 	// compile log parser patterns:
-	var haveError bool
 	for _, parser := range l.parsers {
 		if err := parser.Compile(); err != nil {
-			acc.AddError(err)
-			haveError = true
+			return err
 		}
-	}
-	if haveError {
-		return nil
 	}
 
 	l.wg.Add(1)

--- a/plugins/inputs/logparser/logparser_test.go
+++ b/plugins/inputs/logparser/logparser_test.go
@@ -38,12 +38,8 @@ func TestGrokParseLogFilesNonExistPattern(t *testing.T) {
 	}
 
 	acc := testutil.Accumulator{}
-	logparser.Start(&acc)
-	if assert.NotEmpty(t, acc.Errors) {
-		assert.Error(t, acc.Errors[0])
-	}
-
-	logparser.Stop()
+	err := logparser.Start(&acc)
+	assert.Error(t, err)
 }
 
 func TestGrokParseLogFiles(t *testing.T) {

